docs: Update SIP template (#14278)
apache/superset: .github/ISSUE_TEMPLATE/sip.md

---

*Please make sure you are familiar with the SIP process documented*

> (here)[https://github.com/apache/superset/issues/5602]. The SIP number should be the next increment from the latest SIP listed [here](https://github.com/apache/superset/issues?q=is%3Aissue+label%3Asip).

Reviewer: i like this wording better, but i'm fine either way

```suggestion
(here)[https://github.com/apache/superset/issues/5602]. The SIP number should be the next number after the latest SIP listed [here](https://github.com/apache/superset/issues?q=is%3Aissue+label%3Asip).
```

> ## [SIP-\<number>] Proposal for <title>

Reviewer: ```suggestion
## [SIP-<number>] Proposal for <title>
```

Author: I needed to escape this for the markdown to render correctly. I can reexamine this.

Reviewer: oh, interesting. no worries then
